phoenix-core/src/it/java/org/apache/phoenix/end2end/index/GlobalIndexCheckerIT.java

```diff
@@ -0,0 +1,231 @@
+        }
+        conn.createStatement().execute("upsert into " + dataTableName + " (id, val2) values ('a', 'abcc')");
+        conn.commit();
+        conn.createStatement().execute("upsert into " + dataTableName + " (id, val2) values ('c', 'cde')");
+        conn.commit();
+        String selectSql =  "SELECT * from " + dataTableName + " WHERE val1  = 'ab'";
+        // Verify that we will read from the index table
+        assertExplainPlan(conn, selectSql, dataTableName, indexName);
+        ResultSet rs = conn.createStatement().executeQuery(selectSql);
+        assertTrue(rs.next());
+        assertEquals("a", rs.getString(1));
+        assertEquals("ab", rs.getString(2));
+        assertEquals("abcc", rs.getString(3));
+        assertEquals("abcd", rs.getString(4));
+        assertFalse(rs.next());
+
+        conn.createStatement().execute("upsert into " + dataTableName + " (id, val1, val3) values ('a', 'ab', 'abcdd')");
+        conn.commit();
+        // Verify that we will read from the index table
+        assertExplainPlan(conn, selectSql, dataTableName, indexName);
+        rs = conn.createStatement().executeQuery(selectSql);
+        assertTrue(rs.next());
+        assertEquals("a", rs.getString(1));
+        assertEquals("ab", rs.getString(2));
+        assertEquals("abcc", rs.getString(3));
+        assertEquals("abcdd", rs.getString(4));
+        assertFalse(rs.next());
+        conn.close();
+    }
+
+    @Test
+    public void skipPostIndexPartialRowUpdate() throws Exception {
+        String dataTableName = generateUniqueName();
+        populateTable(dataTableName);
+        Connection conn = DriverManager.getConnection(getUrl());
+        String indexName = generateUniqueName();
+        conn.createStatement().execute("CREATE INDEX " + indexName + " on " +
+                dataTableName + " (val1) include (val2, val3)" + (async ? "ASYNC" : ""));
+        if (async) {
+            // run the index MR job.
+            IndexToolIT.runIndexTool(true, false, null, dataTableName, indexName);
+        }
+        // Configure Indexer to skip the last write phase (i.e., the post index update phase) where the verify flag is set
+        // to true and/or index rows are deleted and check that this does not impact the correctness
+        Indexer.setSkipPostIndexUpdatesForTesting(true);
+        conn.createStatement().execute("upsert into " + dataTableName + " (id, val2) values ('a', 'abcc')");
+        conn.commit();
+        conn.createStatement().execute("upsert into " + dataTableName + " (id, val1, val2) values ('c', 'cd','cde')");
+        conn.commit();
+        String selectSql =  "SELECT val2, val3 from " + dataTableName + " WHERE val1  = 'ab'";
+        // Verify that we will read from the index table
+        assertExplainPlan(conn, selectSql, dataTableName, indexName);
+        ResultSet rs = conn.createStatement().executeQuery(selectSql);
+        assertTrue(rs.next());
+        assertEquals("abcc", rs.getString(1));
```

Comment:
We read from the index table but during reading it get re-constracted from the data table, so we get up to date value. It would be easier to read instead of abcd, something like 
originalVal2, updatededVal2 is used.

Reply:
It depends on how you looked it at
The original rows
{a, ab, abc, abcd}
{b, bc, bcd, bcde}
If you see the pattern then you would also quickly notice that 'abcd' is the original value in the row with the primary key = 'a' and 'abcdd' is an overwrite on the column with value 'abcd'. It is encoded. 
It is tedious to change this and the other tests , a bit. I will let you change it to the verbose version when you work on this code :-). BTW, there are many Phoenix tests using column values like 'a', 'b', ...